### PR TITLE
Don't make the user do our MFA if they have already done the HMPPs Au…

### PR DESCRIPTION
**Pull request by nickmcmahon01**
…th one.

### CODE REVIEW

File: server/routes/login.js [hunk 1]
```diff
@@ -183,11 +183,7 @@ module.exports = () => (router) => {
 
         res.render('pages/two-factor-auth', { authError: false, csrfToken: res.locals.csrfToken })
       } else {
-        req.user.employeeName = await getStaffMemberEmployeeName(
-          userAuthentication.ApiUrl,
-          utilities.getStartMonth(),
-          req.user.token,
-        )
+        req.user.employeeName = jwtDecode(req.user.token).name
```
Comment (astroash) [inline server/routes/login.js:186]: Were we using the property `employeeName` on the last PR. Thought you named it something different. Not sure if that matters

Comment (nickmcmahon01) [reply server/routes/login.js:186]: we were, employeename is the friendly name, not the ID

File: server/routes/login.js [hunk 2]
```diff
@@ -129,8 +125,12 @@ module.exports = () => (router) => {
       }
 
       const quantumAddresses = config.quantumAddresses.split(',')
-
-      if (config.twoFactorAuthOn === 'true' && ipRangeCheck(ipAddress, quantumAddresses) === false) {
+      const hmppsAuthMFAUser = jwtDecode(req.user.token).authorities.includes('ROLE_MFA')
+      if (
+        hmppsAuthMFAUser === false &&
+        config.twoFactorAuthOn === 'true' &&
+        ipRangeCheck(ipAddress, quantumAddresses) === false
```
Comment (jonhindle) [inline server/routes/login.js:132]: I'd question the necessity of `=== false`. If we're having issues with these not being Boolean I'd suggest that code needs looking at.

Comment (nickmcmahon01) [reply server/routes/login.js:132]: "I'd suggest that code needs looking at" I agree.

**Pull request merged**